File: mvp/src/main/java/com/github/wally/mvp/mvp/contract/GankMeiZiListContract.java
package com.github.wally.mvp.mvp.contract;

import com.github.wally.mvp.base.IBaseModel;
import com.github.wally.mvp.base.IBaseView;
import com.github.wally.mvp.base.IPresenter;
import com.github.wally.mvp.bean.gank.GankMeiZiListBean;
import com.github.wally.mvp.http.IDataSource;

import io.reactivex.Observable;

/**
 * Package: com.github.wally.mvp.mvp.contract
 * FileName: GankMeiziContract
 * Date: on 2018/6/16  上午10:35
 * Auther: zihe
 * Descirbe:
 * Email: hezihao@linghit.com
 */
public interface GankMeiZiListContract {
    interface View extends IBaseView {
        void showMeiZiList(IDataSource<GankMeiZiListBean> meiziGankBean, boolean isRefresh);
    }

    interface Presenter extends IPresenter<View> {
        void getMeiZiList(int page, int size, boolean isRefresh);
    }

    interface Model extends IBaseModel {
        Observable<IDataSource<GankMeiZiListBean>> requestMeiZiList(int page, int size);
    }
}

File: mvp/src/main/java/com/github/wally/mvp/mvp/contract/GankMeiZiListContract.kt
package com.github.wally.mvp.mvp.contract

import com.github.wally.mvp.base.IBaseModel
import com.github.wally.mvp.base.IBaseView
import com.github.wally.mvp.base.IPresenter
import com.github.wally.mvp.bean.gank.GankMeiZiListBean
import com.github.wally.mvp.http.IDataSource

import io.reactivex.Observable

/**
 * Package: com.github.wally.mvp.mvp.contract
 * FileName: GankMeiziContract
 * Date: on 2018/6/16  上午10:35
 * Auther: zihe
 * Descirbe:
 * Email: hezihao@linghit.com
 */
interface GankMeiZiListContract {
    interface View : IBaseView {
        fun showMeiZiList(meiziGankBean: IDataSource<GankMeiZiListBean>, isRefresh: Boolean)
    }

    interface Presenter : IPresenter<View> {
        fun getMeiZiList(page: Int, size: Int, isRefresh: Boolean)
    }

    interface Model : IBaseModel {
        fun requestMeiZiList(page: Int, size: Int): Observable<IDataSource<GankMeiZiListBean>>
    }
}